README.md

```diff
@@ -159,33 +159,76 @@ you shouldn't need a binary cache for that.
 
 ## Nix flakes
 
-Note that nix flake support is experimental at the moment.
-Still, there's some minimal usage example.
-You can build an activation package by procuring flake-powered nix
-(`nix run nixpkgs.nixFlakes`,
- `echo "experimental-features = nix-command flakes" >> ~/.config/nix/nix.conf`)
-writing a `flake.nix`:
+**Note:** Nix flake support is still experimental at the moment and subject to change.
+
+Do not run `nix profile` because this will render your environment incompatible with `nix-on-droid` as it relies on
+`nix-env`.
+
+### Minimal example
+
+```nix
+{
+  description = "nix-on-droid configuration";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/release-21.11";
```

Comment:
```suggestion
    nixpkgs.url = "nixpkgs/release-21.11";
```

use the registry :)

Reply:
> If I configure my system registry to let it point nixpkgs to my local revision (a path to a forked repo on my machine), will that be picked up?

I think so if you use `nix registry pin`. Otherwise, I don't see the point of pinning a flake.

> What advantages do we have when using this shorthands?

imho, it's cleaner because it's shorter, and I would tend to trust that the registry is pointing to the right place.

Reply:
also if every flake would use the registry, it would be a lot easier to pin nixpkgs to unstable or something and have all your projects follow it. Like in my case I use flakes for many projects, I think I'll start pinning nixpkgs to unstable.

Reply:
Because these flakes here should show best practices and a good starting point, I prefer to not use shorthand registry identifier.

If you pin nixpkgs on your system to any revision and want to use it for your personal projects, you can do so. If you are just copying these examples with the registry urls, you may use a pinned nixpkgs without knowing. I recently had a problem with a pinned nixpkgs and was searching quite a while before realising that `nixpkgs` input in my flake was referencing a local store path instead of a nixpkgs git revision, which my CI obviously could not find.

So feel free to use registry shortcuts if you know what you are doing, but I would rather not recommend it as a best practice here because it can be tedious to debug.

Reply:
Huh? Registry doesn't pin to commits. Personally, I'd trust the nixpkgs entry value in the registry more than the contents of nixpkgs itself.

Reply:
yeah. pins must be local. in `~/.config/nix/registry.json` or something.

Reply:
You can pin a special revision or a path to a local forked nixpkgs with a registry which then pins to commits. The one thing I trust the most not to have to trust any registry being correct. But as I said: Everyone can use it as he/she likes. This only tries to show a reasonable default that does not need any knowledge about the nix registry. I can add a small paragraph mentioning that it is possible to use registry shortcuts but that would just be a duplication of the nix manual.

Reply:
FWIW, I'm fine with not using registry.